test/autorest/generated/bytegroup/internal/bytegroup/models.go

```diff
@@ -1,53 +1,64 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
 
 package bytegroup
 
 import "github.com/Azure/azure-sdk-for-go/sdk/azcore"
 
-// Error ...
-type Error struct {
-	Status  string
-	Message string
+// ByteGetEmptyResponse contains the response from method Byte.GetEmpty.
+type ByteGetEmptyResponse struct {
+	// StatusCode contains the HTTP status code.
+	StatusCode int
+	// The empty byte value ''
+	Value []byte
 }
 
-func newError(resp *azcore.Response) error {
-	err := Error{}
-	if err := resp.UnmarshalAsJSON(&err); err != nil {
-		return err
-	}
-	return err
+// ByteGetInvalidResponse contains the response from method Byte.GetInvalid.
+type ByteGetInvalidResponse struct {
+	// StatusCode contains the HTTP status code.
+	StatusCode int
+	// Non-ascii base-64 encoded byte string hex(FF FE FD FC FB FA F9 F8 F7 F6)
+	Value []byte
 }
 
-func (e Error) Error() string {
-	return e.Message
+// ByteGetNonASCIIResponse contains the response from method Byte.GetNonASCII.
+type ByteGetNonASCIIResponse struct {
+	// StatusCode contains the HTTP status code.
+	StatusCode int
+	// Non-ascii base-64 encoded byte string hex(FF FE FD FC FB FA F9 F8 F7 F6)
+	Value []byte
 }
 
-// GetEmptyResponse ...
-type GetEmptyResponse struct {
+// ByteGetNullResponse contains the response from method Byte.GetNull.
+type ByteGetNullResponse struct {
+	// StatusCode contains the HTTP status code.
 	StatusCode int
-	Value      []byte `json:"value,omitempty"`
+	// The null byte value
+	Value []byte
 }
 
-// GetInvalidResponse ...
-type GetInvalidResponse struct {
+// BytePutNonASCIIResponse contains the response from method Byte.PutNonASCII.
+type BytePutNonASCIIResponse struct {
+	// StatusCode contains the HTTP status code.
 	StatusCode int
-	Value      []byte `json:"value,omitempty"`
 }
 
-// GetNonASCIIResponse ...
-type GetNonASCIIResponse struct {
-	StatusCode int
-	Value      []byte `json:"value,omitempty"`
+type Error struct {
+	Message string
+	Status int32
 }
 
-// GetNullResponse ...
-type GetNullResponse struct {
-	StatusCode int
-	Value      []byte `json:"value,omitempty"`
+func newError(resp *azcore.Response) error {
+	err := Error{}
+	if err := resp.UnmarshalAsJSON(&err); err != nil {
+		return err
+	}
+	return err
 }
 
-// PutNonASCIIResponse ...
-type PutNonASCIIResponse struct {
-	StatusCode int
+func (e Error) Error() string {
+	return "TODO"
```

Comment:
We need to discuss what to return here.

Reply:
Agreed, I think the message should include information relevant to the specific operation group. We can discuss it on Tuesday if you want.

Reply:
We can include the group+method name, however there should be detailed error info in the fields that was returned from the service that should be included.  Maybe we just include all the fields?  At any rate we can discuss next week.